scripts/closure/closure_compiler.sh

```diff
@@ -0,0 +1,57 @@
+OPTS=(
+  "--compilation_level=ADVANCED_OPTIMIZATIONS"
+  "--js_output_file=build/closure/closure-bundle.js"
+  "--rewrite_polyfills=false"
+
+  # Flags to simplify debugging.
+  "--formatting=PRETTY_PRINT"
+
+  "--externs=dist/zone_externs.js"
+)
+
+OPTS_FAILED=(
+  "--compilation_level=ADVANCED_OPTIMIZATIONS"
+  "--js_output_file=build/closure/closure-bundle.js"
+  "--rewrite_polyfills=false"
+
+  # Flags to simplify debugging.
+  "--formatting=PRETTY_PRINT"
+
+  "--externs=test/closure/zone_externs_failed.js"
+)
+
+closureFlags=$(mktemp)
+echo ${OPTS[*]} > $closureFlags
+
+closureFailedFlags=$(mktemp)
+echo ${OPTS_FAILED[*]} > $closureFailedFlags
+
+# compile closure test source file
+tsc -p .
+# Run the Google Closure compiler java runnable.
+java -jar node_modules/google-closure-compiler/compiler.jar --flagfile $closureFlags 'build/test/closure/zone.closure.js'
```

Comment:
nit: I prefer to have the flagfile be an actual file in this directory, then you don't have to use the fancy bash array construct that few people understand